Lib/dis.py

```diff
@@ -32,25 +32,30 @@ def _try_compile(source, name):
     return c
 
 def dis(x=None, *, file=None, depth=None):
-    """Disassemble classes, methods, functions, generators,
-       asynchronous generators, coroutines, or code.
+    """Disassemble classes, methods, functions, other compiled objects, or code.
 
-    With no argument, disassemble the last traceback.
+       With no argument, disassemble the last traceback.
 
+       Compiled objects currently include generator objects, async generator
+       objects, and coroutine objects, all of which store their code object
+       in a special attribute.
     """
     if x is None:
         distb(file=file)
         return
-    if hasattr(x, '__func__'):  # Method
+    # Extract functions from methods.
+    if hasattr(x, '__func__'):
         x = x.__func__
-    if hasattr(x, '__code__'):  # Function
+    # Extract compiled code objects from...
+    if hasattr(x, '__code__'):  # ...a function, or
         x = x.__code__
-    if hasattr(x, 'gi_code'):  # Generator
+    elif hasattr(x, 'gi_code'):  #...a generator object, or
```

Comment:
At least some of the others are deliberately different because they refer to slightly different things (e.g. cr_await vs gi_yieldfrom).

Anyway, I've added a note about that to the refactoring issue: https://bugs.python.org/issue31197#msg300291